### ISSUE TITLE
Changing RGB lights in kobo aura one now crashes, it didn't do this before

### ISSUE THREAD

**Issue by fluttirci** [koreader/koreader#8899]
* KOReader version: 2022.02
* Device: Kobo aura one

#### Issue
This device has RGB leds, meaning the lights can be any color. Full blue, red , or orange or anything in between. There is a configure menu in the frontlight. In there, I used to be able to change the red green or blue values by pressing + or - . Now I can't. When I press them they crash. However, if I click the textbox, and change the value like that, it won't crash and changing colors work only then. But + - buttons cause a crash. Restore defaults button also makes it crash. I will post images, as you can see my screen is completely red anyway.

Atleast if possible can you tell me what causes it? I am kinda a developer, maybe I could fix that lua file it points to. Seems like a a number variable expects a number, instead it gets a text variable, seen from the errors. But dont know lua much so cant say for sure.
#### Steps to reproduce
Open koreader, click on frontlight, click configure, click red + or - , or restore defaults, it's now crashed.
##### `crash.log` (if applicable)
`crash.log` is a file that is automatically created when KOReader crashes. It can
normally be found in the KOReader directory:

* `/mnt/private/koreader` for Cervantes
* `koreader/` directory for Kindle
* `.adds/koreader/` directory for Kobo
* `applications/koreader/` directory for Pocketbook

Android logs are kept in memory. Please go to Help → Bug Report to save these logs to a file.

Please try to include the relevant sections in your issue description.
You can upload the whole `crash.log` file on GitHub by dragging and
dropping it onto this textbox.

If you instead opt to inline it, please do so behind a spoiler tag:
<details>
  <summary>crash.log</summary>

```
<Paste crash.log content here>

It's a scroll... It's a codex... It's KOReader!

 [*] Current time: 03/13/22-23:09:30
 [*] Version: v2022.02

ffi.load: blitbuffer
03/13/22-23:09:30 INFO  initializing for device Kobo_daylight
03/13/22-23:09:30 INFO  framebuffer resolution: {
    ["w"] = 1404,
    ["h"] = 1872,
}
[ko-input] Forked off fake event generator (pid: 1380)
ffi.load: libs/libmupdf.so
ffi.load: libs/libwrap-mupdf.so
ffi.load: libs/libfreetype.so.6
ffi.load: libs/libharfbuzz.so.0
ffi.load: libs/libzstd.so.1
ffi.load: libs/libutf8proc.so.2
03/13/22-23:09:31 INFO  Loading plugins from directory: plugins
ffi.load: sqlite3
03/13/22-23:10:11 INFO  GestureDetector:probeClockSource: Touch event timestamps appear to use CLOCK_MONOTONIC
./luajit: frontend/util.lua:549: attempt to index local 'text' (a number value)
stack traceback:
	frontend/util.lua:549: in function 'splitToChars'
	frontend/ui/widget/inputtext.lua:381: in function 'initTextBox'
	frontend/ui/widget/inputtext.lua:885: in function 'setText'
	frontend/ui/widget/naturallightwidget.lua:350: in function 'setValueTextBox'
	frontend/ui/widget/naturallightwidget.lua:107: in function 'callback'
	frontend/ui/widget/button.lua:392: in function 'handleEvent'
	frontend/ui/widget/container/inputcontainer.lua:263: in function 'handleEvent'
	frontend/ui/widget/container/widgetcontainer.lua:95: in function 'propagateEvent'
	frontend/ui/widget/container/widgetcontainer.lua:113: in function 'handleEvent'
	frontend/ui/widget/container/widgetcontainer.lua:95: in function 'propagateEvent'
	frontend/ui/widget/container/widgetcontainer.lua:113: in function 'handleEvent'
	...
	frontend/ui/widget/container/widgetcontainer.lua:113: in function 'handleEvent'
	frontend/ui/widget/container/widgetcontainer.lua:95: in function 'propagateEvent'
	frontend/ui/widget/container/widgetcontainer.lua:113: in function 'handleEvent'
	frontend/ui/uimanager.lua:1119: in function 'sendEvent'
	frontend/ui/uimanager.lua:227: in function '__default__'
	frontend/ui/uimanager.lua:1603: in function 'handleInputEvent'
	frontend/ui/uimanager.lua:1683: in function 'handleInput'
	frontend/ui/uimanager.lua:1734: in function 'run'
	./reader.lua:323: in main chunk
	[C]: at 0x0001408d
!!!!
Uh oh, something went awry... (Crash n°1: 03/13/22 @ 23:11:16)
Running FW 4.31.19086 on Linux 3.0.35+ (#5031 PREEMPT Tue Mar 30 14:26:27 CST 2021)
Attempting to restart KOReader . . .
!!!!
```
</details>

**Comment by fluttirci** [comment]
![20220313_231026](https://user-images.githubusercontent.com/83621008/158078213-2071030d-910a-4af9-bf62-aa4260272ee7.jpg)
![20220313_231108](https://user-images.githubusercontent.com/83621008/158078216-04840593-3451-4a6f-b5a2-7067988b2801.jpg)
![20220313_231123](https://user-images.githubusercontent.com/83621008/158078219-8f1419e2-557a-4210-9743-96e0cde68b9a.jpg)

**Comment by Frenzie** [comment]
Pinging @zwim and/or @NiLuJe; I don't really know what you guys have been doing with timestamps.

**Comment by NiLuJe** [comment]
The TS stuff in unrelated, it just happens to be logged on startup regardless of debug logging ;).

As I can attest to, the FL widget is a mess; and IIRC, the RGB widget isn't any better, and is even more hampered by the fact that it's only there for a single device: the Aura One, which severely limits how well we can test it (read: we usually don't. at all :D).

**Comment by NiLuJe** [comment]
Commit above ought to fix it (and allow devs to catch similar mistakes when running in debug code).

**Comment by fluttirci** [comment]
Atleast if possible can you tell me what causes it? I am kinda a developer, maybe I could fix that lua file it points to. Seems like a a number variable expects a number, instead it gets a text variable, seen from the errors. But dont know lua much so cant say for sure.

**Comment by NiLuJe** [comment]
The reverse, actually: the widget is feeding a number to a function ultimately expecting a string. Other *Text widgets already had such guards in place, not sure why this one fell through the cracks and/or why it didn't cause a problem before (IIRC, there were a few Inputext extensions/refactor recently).

**Comment by fluttirci** [comment]
OK IT WORKS NOW YESS! Thank you. I saw your commits there, literally copied that committed 2 files and replaced them, now it works. 
I wanna donate sometime, but with the war going and devaluation of money etc., it won't be much unfortunately. Peace out. Good night :)

**Comment by Frenzie** [comment]
Fixed in #8856.
